### PR TITLE
Add Realtime source archetype

### PR DESCRIPTION
**Pull request by sreevatsanraman**
Archetype for ETL sources and sinks and transforms 
- [x] Realtime source

Upcoming PRs

- [ ] Realtime Sink 
- [ ] Batch Source
- [ ] Batch Sink
- [ ] Transformations

Tests:
a> Install archetype locally
`mvn clean install -DskipTests -P templates`

b> Create realtime source archetype
`mvn -o archetype:generate -DarchetypeRepository=local -DarchetypeGroupId=co.cask.cdap -DarchetypeArtifactId=cdap-etl-realtime-source-archetype -DarchetypeVersion=3.0.0-SNAPSHOT`

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/src/main/java/Source.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+   * Config class for Source.
+   */
+  public static class SourceConfig extends PluginConfig {
+
+    @Name("param")
+    @Description("Source Param")
+    private String param;
+
+
+    public SourceConfig(String param) {
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/src/main/java/Source.java]: Remove this constructor. Better to have default constructor.

Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/src/main/java/Source.java]: Fixed

File: cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/src/main/java/Source.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    this.config = config;
+  }
+
+
+  /**
+   * Config class for Source.
+   */
+  public static class SourceConfig extends PluginConfig {
+
+    @Name("param")
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/src/main/java/Source.java]: Maybe good to have some comment to say only primitives (included boxed types) and String are supported.

File: cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/pom.xml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.felix</groupId>
+          <artifactId>maven-bundle-plugin</artifactId>
+          <version>2.3.7</version>
+          <extensions>true</extensions>
+          <configuration>
+            <instructions>
+              <Embed-Dependency>*;inline=false;scope=compile</Embed-Dependency>
```
Comment (chtyim) [inline cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/pom.xml]: Better include the `Export-Package` as well together with a comment saying that only `@Plugin` classes in the export packages will be included as plugin.

**Comment by chtyim** [comment]
done

File: cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/src/main/java/Source.java (hunk 2 above)
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/src/main/java/Source.java]: Fixed.

File: cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/pom.xml (hunk 3 above)
Comment (sreevatsanraman) [inline cdap-app-templates/cdap-etl/cdap-etl-archetypes/cdap-etl-realtime-source-archetype/src/main/resources/archetype-resources/pom.xml]: Fixed.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**